homeassistant/components/local_todo/todo.py

```diff
@@ -0,0 +1,162 @@
+"""A Local To-do todo platform."""
+
+from collections.abc import Iterable
+import dataclasses
+import logging
+from typing import Any
+
+from ical.calendar import Calendar
+from ical.calendar_stream import IcsCalendarStream
+from ical.store import TodoStore
+from ical.todo import Todo, TodoStatus
+from pydantic import ValidationError
+
+from homeassistant.components.todo import (
+    TodoItem,
+    TodoItemStatus,
+    TodoListEntity,
+    TodoListEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_TODO_LIST_NAME, DOMAIN
+from .store import LocalTodoListStore
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PRODID = "-//homeassistant.io//local_todo 1.0//EN"
+
+ICS_TODO_STATUS_MAP = {
+    TodoStatus.IN_PROCESS: TodoItemStatus.NEEDS_ACTION,
+    TodoStatus.NEEDS_ACTION: TodoItemStatus.NEEDS_ACTION,
+    TodoStatus.COMPLETED: TodoItemStatus.COMPLETED,
+    TodoStatus.CANCELLED: TodoItemStatus.COMPLETED,
+}
+ICS_TODO_STATUS_MAP_INV = {
+    TodoItemStatus.COMPLETED: TodoStatus.COMPLETED,
+    TodoItemStatus.NEEDS_ACTION: TodoStatus.NEEDS_ACTION,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the local_todo todo platform."""
+
+    store = hass.data[DOMAIN][config_entry.entry_id]
+    ics = await store.async_load()
```

Comment:
Maybe we should raise ConfigEntryNotReady if the store fails to load?

Reply:
I've updated this to catch `OSError` and raise `PlatformNotReady` as a first step of exercising this case.

My impression is that if we want this to raise `ConfigEntryNotReady` then we'd want to move this check into the init function. We'd then either load twice (once there and once in the platform) or also pass in the loaded `calendar` object. When I thought through this before when working with local calendar (which this pattern is copied from), it seemed a little unnatural from a code/readability perspective to do all that vs just passing in the store here. 

My impression is that raising `ConfigEntryNotReady` from init has a better error message from the user perspective.

The simplest option seems like it might be to just load twice? (throwing away the results in init).  WDYT?

Also, local calendar needs similar adjustment.